lep/DRAFT/010_Super_Easy_User_Experience.md

```diff
@@ -0,0 +1,217 @@
+  1. A new desktop shortcut is created.
+  1. A new start menu shortcut is created.
+  1. A new Lantern folder is created in the start menu.
+1. After dismissing the message, Lantern is opened.
+1. Lantern adds itself to the system start-up sequence.
+
+### OSX
+
+The OSX installation proccess would be a bit confusing for Windows users, but
+it's usual stuff for OSX users.
+
+I propose not to assume that the user that is going to install Lantern knows
+how to install something in OSX and has no idea about the meaning of the
+Applications directory. I also propose renaming the volume that is created by
+the DMG file into "Lantern Installer" and removing the Lantern icon from it, in
+order to be absolutely clear on the difference between that volume and Lantern
+itself.
+
+1. The user downloads the `.dmg` file. A new file named `lantern-installer.dmg`
+   is created in the `Downloads` folder.
+1. The user double-clicks on the `lantern-installer.dmg` file and a new volume
+   is created on the Desktop. This volume is named "Lantern Installer" and has
+   the icon of a typical volume.
+1. The user double clicks the "Lantern Installer" icon on the desktop and a new
+   window appears. This windows has two icons within, "Lantern" and the
+   "Applications" directory.  There is an arrow in the background and a
+   visibile text states clearly "Drag Lantern into the Applications directory
+   to install".
+1. The user drags the Lantern icon into the Applications directory.
+1. The user starts Lantern by going to the Applications directory and looking
+   for the Lantern icon or by typing "Lantern".
+1. Lantern starts up  and adds itself to the system start-up sequence.
+
+### Ubuntu Linux
+
+I would not change the current Linux installation procedure. Linux users
+usually figure out how to do things by themselves.
+
+## Current day-to-day usage
+
+The Lantern front-end is a web application that behaves the same regardless of
+the operating system.
+
+### Issue #1: IP Address and port in the address bar.
+
+Lantern shows an IP address and port combination in the browser's address bar
+which is unusual to see. The `127.0.0.1` address is due to the need to connect
+to localhost in order to show Lantern's UI. We have two reasons for the odd
+port: the first one if that Lantern should not interfere with any user
+application that may use a common port, in this case the standard port for
+serving web applications is 80; the second reason is that ports lower than 1000
+require administrator privileges at least on UNIX systems.
+
+Proposal: We can get rid of the IP by creating a especial domain name
+http://my.getlantern.org that in case of being blocked it would still be
+reachable through Lantern. The `my.getlantern.org` domain may not be allowed to
+access websocket services on 127.0.0.1 unless a proper
+[CORS](http://www.w3.org/TR/access-control/) setting is used. Unfortunately, we
+won't be able to use https for my.getlantern.org, as the local websocket
```

Comment:
One possibility here @xiam is that we use HTTPS for the client domain and actually ship the private key. This is the only place it would be used, so essentially revealing the public key wouldn't have any consequence beyond using HTTP like we do now. That would allow use to use SSL for both.